libs/wingsdk/src/std/array.ts

```diff
@@ -28,6 +28,30 @@ export class ImmutableArray {
     throw new Error("Abstract");
   }
 
+  /**
+   * Merge arr to the end of this array
+   * @param arr array to merge
+   *
+   * @returns a new ImmutableArray with the values of this array followed by the values of arr
+   */
+  public concat(arr: ImmutableArray): ImmutableArray {
```

Comment:
What should happen if someone tries to concat two mutable arrays? Is the resulting array also mutable?

Reply:
I'd say since wing promotes immutability by default it's ok to always return an immutable array and if the user wants it mutable they can call `.copy()` on the result. Does that make sense?

Reply:
Also noticed that the current implementation requires converting a mutable array to immutable if it is passed as an argument to `.concat()`, which might not be the best DX, but may still be legit:
```ts (wing)
let some_mut_arr   = MutArray<num> [1, 2, 3];
let some_mut_arr2 = MutArray<num> [4, 5, 6];
some_mut_arr.concat(some_mut_arr2.copy());
```

Reply:
I think it may be strange that two mutable things create something immutable, but maybe from design perspective its just a safer choice.

What I would be concerned about then is that this api alone will probably not allow you to pass in a Mutable array without overriding the method in the subclass and changing the arg type to `MutableArray`

Reply:
> What I would be concerned about then is that this api alone will probably not allow you to pass in a Mutable array without overriding the method in the subclass and changing the arg type to MutableArray

Yeah, that's ugly 😄 I will avoid overriding the `.concat()` method in `std.MutableArray`.
I think I'll need to find a way to make our type_checker treat `MutArray` as a subtype of `Array`.

Reply:
Fixed with help from @yoav-steinberg. `MutArray` is considered a subtype of `Array` if their values are subtypes. I'll add the same logic to `Map`/`MutMap` and `Set`/`MutSet`.
Commit: [`73e66ca` (#1751)](https://github.com/winglang/wing/pull/1751/commits/73e66ca89b23d2cd05858ff88f5e9fe2bf9e1425)